ckan/lib/search/query.py

```diff
@@ -89,6 +89,26 @@ def convert_legacy_parameters_to_solr(
     return solr_params
 
 
+def _get_local_query_parser(q: str) -> str:
+    """
+    Given a Solr `q` parameter, extract any custom query parsers used in the
+    local parameters, .e.g. q={!child ...}...
+    """
+    qp_type = ""
+    if not q.startswith("{!"):
```

Comment:
do we need to be more careful with this, e.g. is whitespace before `{!` ignored?